scm-ui/ui-webapp/src/repos/containers/AddRepository.tsx

```diff
@@ -42,65 +43,121 @@ import {
   getNamespaceStrategies,
   isFetchNamespaceStrategiesPending
 } from "../../admin/modules/namespaceStrategies";
+import { RouteComponentProps, withRouter } from "react-router-dom";
+import { compose } from "redux";
+import ImportRepository from "./ImportRepository";
 
-type Props = WithTranslation & {
-  repositoryTypes: RepositoryType[];
-  namespaceStrategies: NamespaceStrategies;
-  pageLoading: boolean;
-  createLoading: boolean;
-  error: Error;
-  repoLink: string;
-  indexResources: any;
-
-  // dispatch functions
-  fetchNamespaceStrategiesIfNeeded: () => void;
-  fetchRepositoryTypesIfNeeded: () => void;
-  createRepo: (
-    link: string,
-    repository: RepositoryCreation,
-    initRepository: boolean,
-    callback: (repo: Repository) => void
-  ) => void;
-  resetForm: () => void;
-
-  // context props
-  history: History;
+type Props = WithTranslation &
+  RouteComponentProps & {
+    repositoryTypes: RepositoryType[];
+    namespaceStrategies: NamespaceStrategies;
+    pageLoading: boolean;
+    createLoading: boolean;
+    error: Error;
+    repoLink: string;
+    indexResources: any;
+
+    // dispatch functions
+    fetchNamespaceStrategiesIfNeeded: () => void;
+    fetchRepositoryTypesIfNeeded: () => void;
+    createRepo: (
+      link: string,
+      repository: RepositoryCreation,
+      initRepository: boolean,
+      callback: (repo: Repository) => void
+    ) => void;
+    resetForm: () => void;
+
+    // context props
+    history: History;
+  };
+
+type State = {
+  importPending: boolean;
 };
 
-class Create extends React.Component<Props> {
+class AddRepository extends React.Component<Props, State> {
```

Comment:
Wouldn't it be easier if we use two separate components for import and create and make more use of composition?